examples/tomo/astra_cuda_3d.py

```diff
@@ -0,0 +1,160 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Example using the ASTRA CUDA for 3D geometries."""
+
+# pylint: disable=invalid-name,no-name-in-module
+
+from __future__ import print_function, division, absolute_import
+import os.path as pth
+from future import standard_library
+standard_library.install_aliases()
+
+# External
+import matplotlib
+matplotlib.use('agg')
+import matplotlib.pyplot as plt
+import numpy as np
+
+# Internal
+from odl import (Interval, Rectangle, FunctionSpace, uniform_discr,
+                 uniform_discr_fromspace, uniform_sampling)
+from odl.tomo import (Parallel3dGeometry, CircularConeFlatGeometry,
+                      HelicalConeFlatGeometry)
+from odl.tomo import (astra_gpu_forward_projector_call,
+                      astra_gpu_backward_projector_call)
+
+
+def save_ortho_slices(data, name, sli):
+    """Save three orthogonal slices of the 3D input volume.
+
+    Parameters
+    ----------
+    data : `DiscreteLp`
+    name : `str`
+    sli : 3-element array-like
+    """
+    # indices for the orthogonal slices
+    x, y, z = np.asarray(sli, int)
+    path = pth.join(
+        pth.join(pth.dirname(pth.abspath(__file__)), 'data'), 'astra')
+
+    data.show('imshow',
+              saveto=pth.join(path, '{}_z{:03d}.png'.format(
+                  name.replace(' ', '_'), z)),
+              title='{} [:,:,{}]'.format(name, z),
+              indices=[slice(None), slice(None), z])
+    data.show('imshow',
+              saveto=pth.join(path, '{}_y{:03d}.png'.format(
+                  name.replace(' ', '_'), y)),
+              title='{} [:,{},:]'.format(name, y),
+              indices=[slice(None), y, slice(None)])
+    data.show('imshow',
+              saveto=pth.join(path, '{}_x{:03d}.png'.format(
+                  name.replace(' ', '_'), x)),
+              title='{} [{},:,:]'.format(name, x),
+              indices=[x, slice(None), slice(None)])
+
+    plt.close('all')
+
+# `DiscreteLp` volume space
+vol_shape = (80, 70, 60)
+discr_vol_space = uniform_discr([-0.8, -0.7, -0.6], [0.8, 0.7, 0.6],
+                                vol_shape, dtype='float32')
+
+# Phantom
+phan = np.zeros(vol_shape)
+sli0 = np.round(0.1 * np.array(vol_shape)).astype(int)
+sli1 = np.round(0.4 * np.array(vol_shape)).astype(int)
+sliz0 = np.round(0.1 * np.array(vol_shape)).astype(int)
+sliz1 = np.round(1.9 * np.array(vol_shape)).astype(int)
+phan[sliz0[0]:sliz1[0], sli0[1]:sli1[1], sli0[2]:sli1[2]] = 1
+
+# Create an element in the volume space
+discr_data = discr_vol_space.element(phan)
+
+# Indices of ortho slices
+vol_sli = np.round(0.25 * np.array(vol_shape))
+save_ortho_slices(discr_data, 'phantom 3d gpu', vol_sli)
+
+# Angles
+angle_offset = 0
+angle_intvl = Interval(0, 2 * np.pi)
+angle_grid = uniform_sampling(angle_intvl, 110, as_midp=False)
+
+# Detector
+dparams = Rectangle([-1, -0.9], [1, 0.9])
```

Comment:
For the tests it is (or was) important to figure out if the dimensions are properly handled. But also for the examples it's maybe also good use unequal lengths because then it is obvious which dimension you're looking at in an image. As a new user you probably don't know which dimension is plotted in which direction. This is often a point of confusion especially if you're are used to Matlab.